src/helm/proxy/clients/huggingface_model_registry.py

```diff
@@ -0,0 +1,94 @@
+from typing import Dict, Optional
+from dataclasses import dataclass
+import re
+from helm.common.hierarchical_logger import hlog
+from helm.proxy.models import (
+    Model,
+    MODEL_NAME_TO_MODEL,
+    TEXT_MODEL_TAG,
+    FULL_FUNCTIONALITY_TEXT_MODEL_TAG,
+)
+
+
+@dataclass(frozen=True)
+class HuggingFaceModelConfig:
+    namespace: Optional[str]
+    """Name of the group or user that owns the model. e.g. 'stanford-crfm'
+
+    May be None if the model (e.g. gpt2) does not have a namespace."""
+
+    model_name: str
+    """Name of the model. e.g. 'BioMedLM'
+
+    Does not include the namespace. ."""
+
+    revision: Optional[str]
+    """Revision of the model to use e.g. 'main'.
+
+    If None, use the default revision."""
+
+    @property
+    def model_id(self) -> str:
+        """Return the model ID.
+
+        Examples:
+        - 'gpt2'
+        - 'stanford-crfm/BioMedLM'"""
+        if self.namespace:
+            return f"{self.namespace}/{self.model_name}"
+        return self.model_name
+
+    @staticmethod
+    def from_string(raw: str) -> "HuggingFaceModelConfig":
+        """Parses a string in the format "[namespace/]model_name[#revision]" to a HuggingFaceModelConfig.
+
+        Examples:
+        - 'gpt2'
+        - 'stanford-crfm/BioMedLM'
+        - 'stanford-crfm/BioMedLM#main'"""
+        pattern = r"((?P<namespace>[^/#]+)/)?(?P<model_name>[^/#]+)(#(?P<revision>[^/#]+))?"
```

Comment:
i'm shocked HF doesn't have a utility for this anywhere, but I can't find it.

Reply:
This is slightly different from Hugging Face convention; they don't _really_ recognize the `#revision` syntax, that's more of a git thing.